src/RulesEngine/RulesCache.cs

```diff
@@ -102,19 +125,19 @@ public WorkflowRules GetWorkFlowRules(string workflowName)
         /// <returns>CompiledRule.</returns>
         public IDictionary<string, RuleFunc<RuleResultTree>> GetCompiledRules(string compiledRulesKey)
         {
-            return _compileRules[compiledRulesKey];
+            return _compileRules[compiledRulesKey].Item1;
```

Comment:
Can you add a null check here? It might throw error when compiled rule does not exist